test/testrunner.py

```diff
@@ -160,14 +166,23 @@ def main():
         if specific_test_type == "sanity":
             pull_dlc_images(all_image_list)
         if specific_test_type == "eks":
-            new_eks_cluster_name = setup_eks_clusters(dlc_images)
-        # Execute dlc_tests pytest command
-        pytest_cmd = ["-s", "-rA", test_path, f"--junitxml={report}", "-n=auto"]
+            eks_cluster_name = setup_eks_clusters(dlc_images)
+            # We expect image URIs of only one framework for EKS tests.
+            framework = "mxnet" if "mxnet" in dlc_images else "pytorch" if "pytorch" in dlc_images else "tensorflow"
```

Comment:
how about making setup_eks_clusters also return the framework name of the just-set-up eks cluster? that way you can avoid this repeated logic

Reply:
I would rather not make the `setup_eks_clusters` function do anything apart from its intended purpose. It would make sense now. But at a later point in time, without this context, understanding its purpose would be harder.

On the other hand, `setup_eks_clusters` has no actual need for the entire list of images except to identify the framework for the cluster it will create. So we can just identify the framework here in `main()`, and set only that as the setup function's parameter.